### PR TITLE
Climber: Feature/climber

### PR DESCRIPTION
**Pull request by Kredcool**
Once this is done we should merge it

### CODE REVIEW

File: rio/subsystems/climber.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        # get Constants
+        constants = getConstants("robot_hardware")
+        self.Climber_const = constants["Climber"]
+
+        # motor configuration
+        self.portClimber = rev.CANSparkMax(1, rev.MotorType.kBrushless)
+        self.starboardClimber = rev.CANSparkMax(2, rev.MotorType.kBrushless)
+
+        self.starboardClimber.follow(self.portClimber, True)
```
Comment (KhanSimeoni) [inline rio/subsystems/climber.py:34]: Follower looks good

**Review by KhanSimeoni — APPROVED** [review]
Its a good start, we'll have to update it later but we can work with this

File: rio/subsystems/climber.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    def climb(self, speed):
+        """
+        The main purpose of the climber
+        to climb
+        """
+
+        self.portClimber.SetReference(10)
+        self.backgroundTimer.hasPeriodPassed(5)
+        self.portClimber.SetReference(10)
```
Comment (KenwoodFox) [inline rio/subsystems/climber.py]: Every 5s?
And timer needs to be spawned in `__init__`

**Review by KenwoodFox — CHANGES_REQUESTED** [review]
Fix background timer.

File: rio/subsystems/climber.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    This class represents the Climber
+    as a whole
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # get Constants
+        constants = getConstants("robot_hardware")
+        self.Climber_const = constants["Climber"]
```
Comment (KenwoodFox) [inline rio/subsystems/climber.py]: Does this exist in the constants yet?

File: rio/subsystems/climber.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+        # motor configuration
+        self.portClimber = CANSparkMax(1, CANSparkMaxLowLevel.MotorType.kBrushless)
+        self.starboardClimber = CANSparkMax(2, CANSparkMaxLowLevel.MotorType.kBrushless)
+
+        self.starboardClimber.follow(self.portClimber, True)
```
Comment (KenwoodFox) [inline rio/subsystems/climber.py:34]: Great!

File: rio/subsystems/climber.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+        self.backgroundTimer = wpilib.Timer()
+        self.backgroundTimer.start()
```
Comment (KenwoodFox) [inline rio/subsystems/climber.py:40]: This is not needed.

File: rio/subsystems/climber.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        self.climbPID.setD(1)
+
+    def climb(self, speed):
+        """
+        The main purpose of the climber
+        to climb
+        """
+
+        self.climbPID.setReference(100, CANSparkMaxLowLevel.ControlType.kPosition)
+        self.backgroundTimer.hasPeriodPassed(5)
```
Comment (KenwoodFox) [inline rio/subsystems/climber.py]: Not needed, do not holdup the updates here. only commands should implement delay

**Review by KenwoodFox — CHANGES_REQUESTED** [review]
- [ ] Check if constants has variables
 - [ ] Implement climber in robotcontainer.py
 - [ ] Implement at least one command to control this subsystem
 - [ ] Remove any loop holdup inside climber.py, if commands want to limit the update speed thats where you would do this

File: rio/constants/robot_hardware.yaml [hunk 7]
```diff
@@ -59,6 +59,29 @@ drivetrain:
       # Ratio of motor rotations to module rotations
       ratio: 0.033              # Updated 2/12/22 by Joe
 
+Climber:
+  portClimber:
+    pose_x: 1 # updated 2/17/22 by keegan
+    pose_y: 1 # updated 2/17/22 by keegan
```
Comment (KenwoodFox) [inline rio/constants/robot_hardware.yaml]: These dont need poses?

File: rio/constants/robot_hardware.yaml [hunk 8]
```diff
@@ -59,6 +59,29 @@ drivetrain:
...
+    drive_id: 9 # updated 2/17/22 by keegan
+    steer_id: 10 # updated 2/17/22 by keegan
```
Comment (KenwoodFox) [inline rio/constants/robot_hardware.yaml]: Pfff, what?!

File: rio/constants/robot_hardware.yaml [hunk 9]
```diff
@@ -59,6 +59,29 @@ drivetrain:
...
+
+  starboardClimber:
+    pose_x: 1 # updated 2/17/22 by keegan
+    pose_y: 1 # updated 2/17/22 by keegan
+    drive_id: 11 # updated 2/17/22 by keegan
+    steer_id: 12 # updated 2/17/22 by keegan
```
Comment (KenwoodFox) [inline rio/constants/robot_hardware.yaml]: We only need two ids, what is this?

**Review by KenwoodFox — CHANGES_REQUESTED** [review]

**Pull request merged**